src/AutoRest.CSharp/Output/Models/Responses/MgmtResponseHeaderGroupType.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Output.Builders;
+using AutoRest.CSharp.Output.Models.Types;
+using AutoRest.CSharp.Utilities;
+
+namespace AutoRest.CSharp.Output.Models.Responses
+{
+    internal class MgmtResponseHeaderGroupType: ResponseHeaderGroupType
+    {
+        private static string[] _knownResponseHeaders = new[]
```

Comment:
Copied code.

Reply:
I was following the pattern, sharing code in most of these places requires more templatization which didn't seem like a route that was desirable. If we do this we can do it in a later PR

Reply:
But you don't generate response header types why do you need it?

Reply:
_knownResponseHeaders on the base type is not used.